From 0dbd09479b0de20c4c3f58519c87fed10c692e8b Mon Sep 17 00:00:00 2001
From: Uti Edun <uti.edun@chainguard.dev>
Date: Wed, 15 Oct 2025 11:58:52 +0000
Subject: [PATCH] fix: update launch script to create service directory and
 adjust permissions. This is because on wolfi, the /etc/service dir is already
 owned by runit and symlinking to that location causes error during package
 build

---
 postgres-appliance/launch.sh | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/postgres-appliance/launch.sh b/postgres-appliance/launch.sh
index 1e3e54f..326344f 100755
--- a/postgres-appliance/launch.sh
+++ b/postgres-appliance/launch.sh
@@ -24,7 +24,7 @@ else
     ln -snf /usr/lib/locale/locale-archive.22 /run/locale-archive
 fi
 
-mkdir -p "$PGLOG" "$PGDATA" "$RW_DIR/postgresql" "$RW_DIR/tmp" "$RW_DIR/certs"
+mkdir -p "$PGLOG" "$PGDATA" "$RW_DIR/postgresql" "$RW_DIR/tmp" "$RW_DIR/certs" "$RW_DIR/service"
 if [ "$(id -u)" -ne 0 ]; then
     sed -e "s/^postgres:x:[^:]*:[^:]*:/postgres:x:$(id -u):$(id -g):/" /etc/passwd > "$RW_DIR/tmp/passwd"
     cat "$RW_DIR/tmp/passwd" > /etc/passwd
@@ -46,7 +46,7 @@ for i in $(seq 0 7); do
         done
     fi
 done
-chown -R postgres: "$PGROOT" "$RW_DIR/certs"
+chown -R postgres: "$PGROOT" "$RW_DIR/certs" "$RW_DIR/postgresql" "$RW_DIR/service"
 chmod -R go-w "$PGROOT"
 chmod 01777 "$RW_DIR/tmp"
 chmod 0700 "$PGDATA"
@@ -67,6 +67,14 @@ sv_stop() {
     sv -w 86400 stop /etc/service/*
 }
 
+rm -rf /etc/service
+ln -s "$RW_DIR/service" /etc/service
+
+rm -fr /var/spool/cron /var/tmp
+mkdir -p /var/spool
+ln -s "$RW_DIR/cron" /var/spool/cron
+ln -s "$RW_DIR/tmp" /var/tmp
+ln -s "$RW_DIR/postgres.yml" "$PGHOME/postgres.yml"
 [ ! -d /etc/service ] && exit 1  # /etc/service has not been created due to an error, the container is no-op
 
 trap sv_stop TERM QUIT INT
-- 
2.51.0

